lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1005,6 +1007,62 @@ def invocation_step(self, trans, invocation_id, step_id, **kwd):
         )
         return self.__encode_invocation_step(trans, invocation_step)
 
+    def _generate_invocation_metrics(self, trans, invocation_id, **kwd):
+        """
+        build a dictionary of invocation metrics
+        """
+        decoded_workflow_invocation_id = self.decode_id(invocation_id)
+        workflow_invocation = self.workflow_manager.get_invocation(trans, decoded_workflow_invocation_id)
+        history = workflow_invocation.history
+
+        metrics, h_metrics = {}, {}
+        h_contents = self.history_contents_manager.contained(history)
+
+        for i, step in enumerate(workflow_invocation.steps):
```

Comment:
I guess it would. I have always thought about these things in a 'historical' sense so that did not occur to me. How would you do that? Set an error if the invocation has not finished? Or set it to wait? Not sure how to do that best.